open-grant-proposals/open-proposal-xdv.md

```diff
@@ -0,0 +1,378 @@
+
+To listen and execute next flow, the claim will require additionally integration linked to brokers, issuer (insurance), policy holders agents (VC agent) and other offchain data sources. 
+
+- `Claim` is filed thru decentralized automated legal tech (`broker`) and sent to insurance company.
+- `Policy` is verified by back office and updated in centralized solution. `Broker` uses XDV Protocol oracle contracts
+- After oracle come to a consensus, an event  triggers  a `smart data contract` and orchestrate trusted data which then is computed and notified to subscribers
+
+#### EDI Orders and Invoicing
+
+XDV protocol integrates EDI orders with offchain workers. The grocery consortium each one has different types of data structures, they are looking for:
+
+- Verify EDI orders signed with RSA
+- Convert them to electronic invoice and cloned to XDV Node XDV Storage and configure with RBAC/ACL
+- Push invoice to government tax office with government mandated signature using `smart data contracts`
+
+#### COVID-19 verifiable data warehousing
+
+In this use case, you have a pharmaceutical company, government and patients. To keep the diverse data verifiable and able to be query and used for other, non transactional uses, and because cost savings is very important in these implementations, they decide to use XDV Node and implement XDV protocol as following:
+
+- Pharma company has their vaccine data in Ethereum private blockchain, because is private, decentralized chain, it can be thought as a sidechain. These transactions can be duplicated in XDV Node, and keep the data with selective disclosure or encrypted using XDV or IPFS Storage. 
+- Government keeps  its data in CSV Tabular format. They would like to keep these as usual, with XDV Node, because it uses IPLD Schemas, the government selects the codec format for JSON Table. Any request from the data will be render and store as JSON Table. For this to work, they'll need to use oracle which will sync with XDV Node. Because each oracle will use BLS to sign for proof, it should be more secure than other providers.
+- Patients get mobile and desktop DApps to keep track of news and events.
+
+Pharma and Government, besides the transactional integration which keeps accountability of vaccine success, can also create `smart data contracts`. A `smart data contract` aggregates, orchestrate and maps linked, verifiable data from the chain transactions and connected. 
+
+
+- What are the risks if you don't get it right?
+none
+
+- What are the risks that will make executing on this project difficult?
+finding knowledgable IPLD engineers
+
+
+
+## Deliverables
+
+
+### Metadata / File storage rust-ipld
+
+#### Spec
+
+Create a multiformat `xdv` and migrate PoC to XDV Node (Go) and/or Rust. The API must be 100% compatible with IPFS, ie `POST /add` and other relevant REST methods. It must have an interface with events to register `cross chain swaps` events.
```

Comment:
What do you mean by "multiformat `xdv`". Would you need a new multiformat codec? Earlier in the document you mention that you'll use `dag-cbor`.

> The API must be 100% compatible with IPFS, ie `POST /add` and other relevant REST methods.

Is the idea to just be a proxy and use a full IPFS node under the hood, or re-implementing those. Also "relevent REST methods" is pretty vague, the IPFS API is pretty large, which ones did you have in mind? :)

Reply:
1. sorry,  no need to do new codec
2. only Files (add, get)